signer/src/transaction_coordinator.rs

```diff
@@ -1674,76 +1697,295 @@ where
-                accept_threshold,
+                params.signature_threshold,
             )
             .await?;
 
-        let mut deposits: Vec<utxo::DepositRequest> = Vec::new();
+        // If we didn't find any pending withdrawal requests, we can exit early.
+        if pending_withdraw_requests.is_empty() {
+            tracing::debug!("no pending withdrawal requests eligible for consideration found");
+            return Ok(eligible_withdrawals);
+        }
 
-        for req in pending_deposit_requests {
+        // Iterate over the pending withdrawal requests we fetched above and
+        // validate them against the remaining consensus rules.
+        for req in pending_withdraw_requests {
+            if req.bitcoin_block_height < min_soft_bitcoin_height {
+                tracing::debug!(
+                    request_id = req.request_id,
+                    bitcoin_block_height = req.bitcoin_block_height,
+                    min_soft_bitcoin_height,
+                    reason = SKIP_REASON_SOFT_EXPIRY,
+                    message = REQUEST_SKIPPED_MESSAGE
+                );
+                continue;
+            }
+
+            // [8] Ensure that the withdrawal request amount is above the dust
+            // limit.
+            if req.amount <= WITHDRAWAL_DUST_LIMIT {
+                tracing::debug!(
+                    request_id = req.request_id,
+                    amount = req.amount,
+                    reason = SKIP_REASON_AMOUNT_IS_DUST,
+                    message = REQUEST_SKIPPED_MESSAGE
+                );
+                continue;
+            }
+
+            // [9] Ensure that the withdrawal request amount is within the
+            // current sBTC caps.
+            let per_withdrawal_cap = params.sbtc_limits.per_withdrawal_cap().to_sat();
+            if req.amount > per_withdrawal_cap {
+                tracing::debug!(
+                    request_id = req.request_id,
+                    amount = req.amount,
+                    per_withdrawal_cap = per_withdrawal_cap,
+                    reason = SKIP_REASON_PER_WITHDRAWAL_CAP_EXCEEDED,
+                    message = REQUEST_SKIPPED_MESSAGE
+                );
+                continue;
+            }
+
+            // Calculate the number of blocks passed (confirmations) since the
+            // bitcoin anchor of the stacks block confirming the withdrawal
+            // request.
+            let num_confirmations = params
+                .bitcoin_chain_tip
+                .block_height
+                .saturating_sub(req.bitcoin_block_height);
+
+            // [3] Ensure that we have the required number of confirmations for
+            // the withdrawal request.
+            if num_confirmations < min_confirmations.into() {
+                tracing::debug!(
+                    request_id = req.request_id,
+                    num_confirmations,
+                    required_confirmations = min_confirmations,
```

Comment:
I think this was changed by accident. Shouldn't it be:
```suggestion
            if num_confirmations < WITHDRAWAL_MIN_CONFIRMATIONS.into() {
                tracing::debug!(
                    request_id = req.request_id,
                    num_confirmations,
                    required_confirmations = WITHDRAWAL_MIN_CONFIRMATIONS,
```

Reply:
no it was on purpose, the value (`WITHDRAWAL_MIN_CONFIRMATIONS`) is passed in from `get_pending_requests()` to keep constants out of this method -- I prefer to have the constants "higher up" and passed down to the "worker methods"; it makes the method more flexible which is especially nice in tests.

Reply:
which reminds me, there's still `WITHDRAWAL_DUST_LIMIT`. Do we need both this one and `DEPOSIT_DUST_LIMIT` which are the same value? Could they be consolidated into just `SBTC_DUST_LIMIT` and then it would fit perfectly into `GetPendingRequestsParams`.

Reply:
I misread this before, it looked like `if num_confirmations < num_confirmations.into()`, which I thought was the same variable being used accidentally 😅. I hastily drafted a comment instead of pausing to take a closer look, which is what I would normally do.